core/jvm-native/src/main/scala/cats/effect/unsafe/PollingSystem.scala

```diff
@@ -17,6 +17,23 @@
 package cats.effect
 package unsafe
 
+/**
+ * Encapsulates the methods required for managing and interacting with a polling system. Polling
+ * systems are typically used in scenarios such as handling asynchronous I/O or other
```

Comment:
So this is a really interesting one, depending on what precisely we mean by "asynchronous" :)

_Technically_ if you have truly async I/O I would say you _don't_ need polling, because the I/O is already async and you are just waiting for a callback from somewhere.

Arguably, polling is actually a form of _blocking_ I/O, because you have to block on the poll method. The magic is that (1) we can multiplex polling of multiple sockets into a single blocking call and (2) by integrating this blocking directly into the runtime worker loop it actually becomes "free" i.e. the runtime would have had to block _anyway_ and the user gets an experience of 100% async, non-blocking I/O.

So I wonder if we can use a term like "multiplexed blocking I/O". I just made this up haha so I'm open to better ideas.
```suggestion
 * systems are typically used in scenarios such as handling multiplexed blocking I/O or other
```

Reply:
I find the distinction between truly async I/O and polling operations very interesting and clarifying. 

I must say that the term 'multiplexed blocking I/O' really resonates with the concept. I like it :)